dev/linearhooks/internal/handlers/handlers.go

```diff
@@ -0,0 +1,160 @@
+
+	gqlClient graphql.Client
+}
+
+func New(ctx context.Context, logger log.Logger, b []byte, apiKey, signingSecret string) (*Handler, error) {
+	var c Config
+	if err := yaml.Unmarshal(b, &c); err != nil {
+		return nil, err
+	}
+	if err := c.Validate(); err != nil {
+		return nil, err
+	}
+
+	gqlClient := lineargql.NewGraphQLClient("https://api.linear.app/graphql", apiKey)
+
+	return &Handler{
+		logger:        logger,
+		c:             c,
+		signingSecret: signingSecret,
+		gqlClient:     gqlClient,
+	}, nil
+}
+
+func (h *Handler) HandleIssueMover(w http.ResponseWriter, r *http.Request) {
+	body, err := h.validatePayloadSignature(r)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	eventType := r.Header.Get("Linear-Event")
+	eventId := r.Header.Get("Linear-Delivery")
+	logger := h.logger.With(log.String("event.type", eventType), log.String("event.id", eventId))
+	if eventType != "Issue" && eventType != "Label" {
+		logger.Warn("ignoring unexpected event type")
+		return
+	}
+
+	var e linearschema.Event
+	if err := json.Unmarshal(body, &e); err != nil {
+		logger.Error("decode request body", log.Error(err))
+		_, _ = w.Write([]byte("error decoding request body"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	logger = logger.With(log.String("event.action", string(e.Action)))
+
+	if e.IssueData == nil {
+		logger.Warn("missing issue data in event payload")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	logger = logger.With(log.String("issue.id", e.IssueData.Identifier))
+	logger.Debug("processing issue data change event")
+
+	switch e.Action {
+	case linearschema.ActionTypeCreate, linearschema.ActionTypeUpdate:
+	case linearschema.ActionTypeDelete:
+		// no-op
+		return
+	default:
+		logger.Warn("ignoring unexpected action type", log.String("action", string(e.Action)))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if err := moveIssue(logger, h.gqlClient, h.c.Spec.Mover.Rules, *e.IssueData); err != nil {
+		logger.Error("move issue", log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Write([]byte("OK"))
+	w.WriteHeader(http.StatusOK)
+}
+
+func moveIssue(logger log.Logger, client graphql.Client, rules []RuleSpec, issue linearschema.IssueData) error {
+	for _, r := range rules {
+		if newTeamId := r.moveToNewTeam(issue); newTeamId != nil {
+			logger.Info("moving issue",
+				log.String("from.id", issue.Team.Key),
+				log.String("to", r.Dst.TeamID),
+			)
+			teamUUID, err := getTeamUUID(context.Background(), client, r.Dst.TeamID)
+			if err != nil {
+				return errors.Wrapf(err, "get dst team %q UUID", r.Dst.TeamID)
+			}
+			if err := moveIssueToTeam(context.Background(), client, issue.Identifier, teamUUID); err != nil {
+				return errors.Wrapf(err, "move issue %q to team %q", issue.Identifier, teamUUID)
+			}
+			return nil
+		}
+	}
+	return nil
+}
+
+var (
+	// teamKeyUUIDCache is a cache of team key to team UUID mappings with a 5 minute TTL.
```

Comment:
```suggestion
	// teamKeyUUIDCache caches team identifier -> UUID mappings to avoid
	// re-requesting the same data for every label change.
	// The cache has a TTL for eventually correct behavior in case a team is
	// deleted and re-created.
```

I think it would be more useful to elaborate on the why (why have a cache here? why use a TTL?).

Also, it seems like the terms 'identifier' and 'key' are essentially equivalent in the API, so perhaps we should just consistently use 'identifier'?

Reply:
linear has not been very consistent on theirs own as well. l'll leave a note of `identifier/key -> UUID` to avoid renaming a bunch of things everywhere.